driver/src/test/scala/Common.scala

```diff
@@ -167,4 +166,45 @@ object Common {
 
     slowProxy.stop()
   }
+
+}
+
+trait CommonAuth {
+
+  import reactivemongo.api.AuthenticationMode
+
+  def driver: MongoDriver
+
+  def authMode: Option[AuthenticationMode] =
+    Option(System.getProperty("test.authMode")).flatMap {
+      case "cr"   => Some(CrAuthentication)
+      case "x509" => Some(X509Authentication)
+      case _      => None
+    }
+
+  def defaultAuthentications: Seq[Authenticate] =
+    ifX509(Seq(Authenticate("", certSubject, "")))(otherwise = Nil)
+
+  def certSubject: String =
+    Option(System.getProperty("test.clientCertSubject"))
+      .getOrElse(sys.error("Client cert subject required if X509 auth enabled"))
+
+  def increaseTimeoutIfX509(timeout: FiniteDuration): FiniteDuration =
+    ifX509(timeout * 10)(otherwise = timeout)
+
+  def makeConnection(nodes: Seq[String], options: MongoConnectionOptions): MongoConnection = {
```

Comment:
I don't see benefit for those `makeConnectionX` which are essentially oneliner aliases

Reply:
Renamed to overloaded makeConnetion in both cases. The reasons for having both is that some tests create a separate instance of a driver and control its lifecycle

Reply:
Anyway would directly call `driver.connection` rather than introducing such aliases

Reply:
It's not about aliases here thought. Inside of `Common` you could, but many tests obtain connection not by using `connection` variable but instead use `driver.connection(...)` and modify some parameters. These tests would have to know about `MONGO_PROFILE` being potentially set to `x509` and explicitly add authentication. Here they work the same with or without `x509`

Reply:
For me it's aliasing, a also the call for such functions can be reverted back to `driver.connection`, the default connection in `Commons` being the only place where the test profile is known.